aptos-move/aptos-vm/src/data_cache.rs

```diff
@@ -128,9 +128,21 @@ impl<'e, E: ExecutorView> StorageAdapter<'e, E> {
         address: &AccountAddress,
         struct_tag: &StructTag,
         metadata: &[Metadata],
+    ) -> Result<(Option<Bytes>, usize), VMError> {
```

Comment:
do we still need get_any_resource? If we are consistent with the other changes here, it seems we should get rid of it and make it the caller's responsibility to call get_any_resource_with_layout with None? (harder to accidentally use the "wrong" one later?)